src/time.cr

```diff
@@ -1359,9 +1359,20 @@ struct Time
 
   # Returns a copy of this `Time` representing the beginning of the week.
   #
+  # The week starts on Monday by default, but can be configured by passing a different `start_day` as a `Time::DayOfWeek`.
+  #
+  # ```
+  # now = Time.local                                  # => 2023-05-08 16:48:13 +00:00
+  # now.at_beginning_of_week(Time::DayOfWeek::Monday) # => 2023-05-08 00:00:00 +00:00
+  # now.at_beginning_of_week(Time::DayOfWeek::Sunday) # => 2023-05-07 00:00:00 +00:00
+  #
+  # # with a symbol
+  # now.at_beginning_of_week(:monday) # => 2023-05-08 00:00:00 +00:00
+  # now.at_beginning_of_week(:sunday) # => 2023-05-07 00:00:00 +00:00
+  # ```
   # TODO: Ensure correctness in local time-line.
```

Comment:
IMO there's no need to separately showcase the autocasting feature, I'd suggest to simply show it's possible.

```suggestion
  # ```
  # now = Time.local                  # => 2023-05-08 16:48:13 +00:00
  # now.at_beginning_of_week(:monday) # => 2023-05-08 00:00:00 +00:00
  # now.at_beginning_of_week(:sunday) # => 2023-05-07 00:00:00 +00:00
  # ```
  #
  # TODO: Ensure correctness in local time-line.
```

Reply:
Thank you! I've addressed this here: [`a08f0d2` (#13446)](https://github.com/crystal-lang/crystal/pull/13446/commits/a08f0d22e4832fc5b8dfb395406eac96269007d4)

I also opted to add an example showing the default behaviour without an argument :~)